src/core/DrumSynth.cpp

```diff
@@ -345,7 +337,7 @@ int DrumSynth::GetDSFileSamples(QString dsfile, int16_t*& wave, int channels, sa
 	}
 
 	// try to read version from input file
-	strcpy(sec, "General");
+	std::memcpy(sec, "General", sizeof("General"));
```

Comment:
I do not believe this is correct. `sizeof("General") == sizeof(char*)` which, does not equal `strlen("General")`.

```suggestion
	strncpy(sec, "General", strlen("General") + 1);
```

Reply:
Checking with the standard, I stand corrected:

From C++17 standard, 8.5.2.3, expr.sizeof:

> When applied to an array, the result is the total number of bytes in the array. This implies that the size of an array of n
elements is n times the size of an element.

Reply:
I guess perhaps this was changed in C++11? Hm..
![image](https://github.com/user-attachments/assets/63b326d2-0251-4c7d-ad3d-b15eafcc7823)